### PR TITLE
Load More Button On Profile Page

### PR DESCRIPTION
**Pull request by GrahamSH-LLK**
@WorldLanguages This is the reopened PR.

### CODE REVIEW

**Comment by GrahamSH-LLK** [comment]
@TheColaber Here you go!

**Comment by Hans5958** [comment]
Mind the capitalization?

**Review by WorldLanguages — CHANGES_REQUESTED** [review]
1) You should not get the username from the HTML. This would break on ST profiles bc of the asterisk.
2) When clicking "load more", it scrolls to the bottom automatically, exactly like explore used to do before we turned on our addon fix, and the ST eventually fixed it.
3) You should only request the activity once. If the user clicks load more again, you should retrieve the data from a variable.

File: addons/full-signature/working.css [hunk 1]
```diff
@@ -5,3 +5,9 @@
 ul.activity-stream div {
   width: 90%;
 }
+
+.load-more-wibd {
+  margin-left: 25%;
+  margin-right: 25%;
```
Comment (WorldLanguages) [inline addons/full-signature/working.css:11]: Also, can you find a way to center the button properly?

Comment (mxmou) [reply addons/full-signature/working.css:11]: It's already better than what other 2.0-styled pages do (`width: auto; margin-left: 45%`).

Comment (WorldLanguages) [reply addons/full-signature/working.css:11]: lol really? that's not nice for internationalization.

Comment (WorldLanguages) [reply addons/full-signature/working.css:11]: I'd prefer to have the button wrapped in a div so that it can be centered properly tbh

**Comment by TheColaber** [comment]
Ok, I fixed all the things above. @GrahamSH-LLK if you could commit these please:


For working.js:
```javascript
export default async function ({ addon, global, console }) {
  let activityStream = document.querySelectorAll(".activity-stream li");
  let container = document.querySelector(".activity-stream").appendChild(document.createElement("div"));
  container.classList.add("load-more-wibd-container");
  let loadMore = container.appendChild(document.createElement("button"));
  loadMore.classList.add("load-more-wibd");
  loadMore.innerText = "Load More";
  let dataLoaded = 6;
  fetch(`
    https://scratch.mit.edu/messages/ajax/user-activity/?user=${
      window.location.pathname.substring(7).slice(0, -1)
    }&max=1000000`)
    .then((response) => response.text())
    .then((response) => {
      let html = new DOMParser().parseFromString(response, "text/html");
      loadMore.addEventListener("click", function () {
        let lastScroll = document.querySelector(".activity-stream").scrollTop
        html.querySelectorAll("ul > li").forEach((li, index) => {
          if (index > dataLoaded && index < dataLoaded + 6) {
            activityStream[activityStream.length - 1].append(li);
          }
        });
        dataLoaded += 6;
        document.querySelector(".activity-stream").scrollTop = lastScroll;
        if (dataLoaded >= html.querySelectorAll("ul > li").length) {
          loadMore.remove()
        }
      });
    });
}
```


For working.css:
```css
.activity-stream {
  overflow-y: scroll !important;
}

ul.activity-stream div {
  width: 90%;
}

.load-more-wibd {
  outline: none;
}

.load-more-wibd-container {
  text-align: center;
}
```

**Comment by GrahamSH-LLK** [comment]
> Ok, I fixed all the things above. @GrahamSH-LLK if you could commit these please:
> 
> For working.js:
> 
<details> <summary></summary>
> ```js
> export default async function ({ addon, global, console }) {
>   let activityStream = document.querySelectorAll(".activity-stream li");
>   let container = document.querySelector(".activity-stream").appendChild(document.createElement("div"));
>   container.classList.add("load-more-wibd-container");
>   let loadMore = container.appendChild(document.createElement("button"));
>   loadMore.classList.add("load-more-wibd");
>   loadMore.innerText = "Load More";
>   let dataLoaded = 6;
>   fetch(`
>     https://scratch.mit.edu/messages/ajax/user-activity/?user=${
>       window.location.pathname.substring(7).slice(0, -1)
>     }&max=1000000`)
>     .then((response) => response.text())
>     .then((response) => {
>       let html = new DOMParser().parseFromString(response, "text/html");
>       loadMore.addEventListener("click", function () {
>         let lastScroll = document.querySelector(".activity-stream").scrollTop
>         html.querySelectorAll("ul > li").forEach((li, index) => {
>           if (index > dataLoaded && index < dataLoaded + 6) {
>             activityStream[activityStream.length - 1].append(li);
>           }
>         });
>         dataLoaded += 6;
>         document.querySelector(".activity-stream").scrollTop = lastScroll;
>         if (dataLoaded >= html.querySelectorAll("ul > li").length) {
>           loadMore.remove()
>         }
>       });
>     });
> }
> ```
> 
> For working.css:
> 
> ```css
> .activity-stream {
>   overflow-y: scroll !important;
> }
> 
> ul.activity-stream div {
>   width: 90%;
> }
> 
> .load-more-wibd {
>   outline: none;
> }
> 
> .load-more-wibd-container {
>   text-align: center;
> }
> ```
></details>
Got it!

**Comment by TheColaber** [comment]
Awesome!!

**Comment by WorldLanguages** [comment]
@TheColaber Why did you make it so that it would load all the activity, even if the user didn't click "load more"? I clearly said:

> You should only request the activity once. If the user clicks load more **again**, you should retrieve the data from a variable.

**Comment by TheColaber** [comment]
> @TheColaber Why did you make it so that it would load all the activity, even if the user didn't click "load more"? I clearly said:
> 
> > You should only request the activity once. If the user clicks load more **again**, you should retrieve the data from a variable.

I'm sorry, I'm so dumb, I wasn't reading... you want me to redo it?

**Comment by WorldLanguages** [comment]
@TheColaber I'll do it

**Comment by TheColaber** [comment]
K.

**Pull request merged**